client/extensions/woocommerce/state/ui/products/test/edits-reducer.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import reducer from '../edits-reducer';
+
+import {
+	editProduct,
+	editNewProduct,
+} from '../actions';
+
+describe( 'edits-reducer', () => {
+	it( 'should initialize to null', () => {
+		expect( reducer( undefined, { type: '@@test/INIT' } ) ).to.equal( null );
+	} );
+
+	it( 'should create "updates" upon first edit', () => {
+		const product = { id: 1 };
+		const edits = reducer( undefined, editProduct( product, { name: 'A Product' } ) );
```

Comment:
How do you feel about using `deepFreeze` in test to ensure reducers create a new state?